base/loading.jl

```diff
@@ -2583,69 +2622,86 @@ function parse_cache_header(f::IO)
     l = read(f, Int32)
     clone_targets = read(f, l)
 
-    return modules, (includes, requires), required_modules, srctextpos, prefs, prefs_hash, clone_targets, flags
+    # determine path for @depot replacement from srctext files only, e.g. ignore any include_dependency files
+    srcfiles = srctext_files(f, srctextpos, includes)
+    depot = resolve_depot(srcfiles)
+    keepidx = Int[]
+    for (i, chi) in enumerate(includes)
+        chi.filename ∈ srcfiles && push!(keepidx, i)
+    end
+    if isnothing(depot)
+        @debug "Failed to determine depot from srctext files in cache file $cachefile."
```

Comment:
Are we passing the `cachefile` argument in only for this error message?

Reply:
> Are we passing the cachefile argument in only for this error message?

Yes. I thought it could be useful for debugging in case one wants to manually inspect a `.ji` file.
You want me to remove it?

> Should we instead raise an error here?

Will do.

Reply:
Hmmm, when I convert this to an error it throws for `Pkg` during `make`.
I wonder if this has something to do with the recent excision? Although resolving works for `REPL`.

Reply:
> I am also seeing these weird infos We haven't cleaned this depot up for a bit, running Pkg.gc()... during build.

That is known and can be ignored.

Reply:
> Should we instead raise an error here?

I added a `throw(ArgumentError(...))` and it can build now again.